src/components/AppHeader.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+    <header class="app-header">
+        <div class="app-header__item">
+            <img @click="homepageRedirect" src="@/assets/images/logo.svg" alt="monterail logo"/> 
+        </div>
+        <div class="app-header__item desktop">
+            <nav class="navbar">
+                <router-link :to="{ name: 'movies' }" class="navbar__link">
+                    Movies
+                </router-link>
+                <router-link :to="{ name: 'screenings' }" class="navbar__link">
+                    Screenings
+                </router-link>
+                <router-link :to="{ name: 'contact' }" class="navbar__link">
+                    Contact&nbsp;us
+                </router-link>
+            </nav>
+            <div>
+                <router-link :to="{ name:'register' }">
+                    <base-button >Register</base-button>
+                </router-link>
+                <router-link :to="{ name:'login' }">
+                    <base-button buttonType="primary">Login</base-button>
+                </router-link>
+                
+            </div>
+        </div> 
+        <div class="app-header__item hamburger">
+            <div></div>
+            <div></div>
+            <div></div>
+        </div>
+    </header>
+</template>
+
+<script>
+export default {
+    methods: {
+        homepageRedirect() {
+            this.$router.currentRoute.name != "home" ? this.$router.push({name: "home"}) : ''
```

Comment:
tip: there is an alias for `this.$router.currentRoute` as this is often used in vue applications: `this.$route`